README.md

```diff
@@ -81,6 +81,23 @@ You can make a debug build with `.\make.ps1 build -Config Debug`
 
 ## Create a project with dependencies
 
+### Using Corral for first time?
```

Comment:
I don't think this quite the right language. It isn't about using corral for the first time. It's setting up a project to use corral for the first time.

I'm not sure this is the right approach.

I think a getting started using corral or how to use corral section would be better.

Rather than adding this one smaller addition,

I think detailing the different commands like `init`, `add` etc would be a better approach. From that detailing, there would also be "starting from zero, here is how you use corral".

That's a bit more work than this PR. I would imagine a day or two with thought and revision before opening a PR.

Would you be up for plotting that out and opening a PR for it @adeel41 ?

I'm happy to assist when you run into problems.

Reply:
sounds like a plan. I will revert these changes. Do you think there should be a separate `.md` file for these details? or shall we create a new section between `Building from Source` and `Creating a project with dependencies`?

Reply:
I think for now, separate sections of the README are fine.

We can break them out later if we decide it's needed.

Reply:
@SeanTAllen can you have a look. As per your suggestion, I've added a new section instead of fitting everything in the existing section. I've also added some explanation for the commands which we will be using.

Reply:
Good stuff. I've left some comments. There are a few. I wanted to note there are many comments on smaller things because I think the core of this is very good.